### ISSUE TITLE
how to share labels

### ISSUE THREAD

**Issue by dhejyuyr** [snwagh/falcon-public#54]
hi! Im a newbie in mpc and ppml. I want to test Falcon training and predicting with tabular data (26 columns + 1 label column with 4 values 0-3). As far as I know, the convolutional layers here are tied to the size of the image, so I settled on 3 fully connected layers (input, hidden and output). It's SecureML, but different layers sizes. And a similar model from `MNIST_SecureML.ipynb` on my data with a learning rate `0.001`  and this setup works ok
```
    self.layer 1 = nn.Linear(26, 11, bias=True)
    self.layer2 = nn.Linear(11, 11, bias=True)
    self.layer3 = nn.Linear(11, 4, bias=True)
```

Here are the steps to pre-process my data to falcon:
1) scaled to [0, 1]
2) uploaded them using `floatToMyType` to a vector `data`
3) using `get64Bits` I got 2 random values `r_1, r_2` and `myType r_3 = data[i] - r_1 - r_2`
4) save `r1` to `party_a` vector,  `r2` to `party_b`, `r3` to `party_c`. 
5) save `party_a` vector to party_a file etc.

I save them as `myType` . So when I will upload them to `inputData` I just remove `floatToMyType` cast when reading it in `secondary.cpp`. accordingly, they are loaded as `myType` and there are no problems with this.

But I don't understand how to share labels.
I have 4 labels: 
0, 1, 2, 3.
If I use one-hot encoding, then this is
1 0 0 0
0 1 0 0
0 0 1 0
0 0 0 1

I need to do share(1),share(0),share(0),share(0) and so on for each row? Or do I need to do something else?

**Comment by dhejyuyr** [comment]
i was right :)
